### PR TITLE
feat(translator): implement idle timeout in ClientTrafficPolicy

### LINKED ISSUE #2861: feat: add support to configuring downstream idle timeout
*Description*:
Add `idleTimeout` to clientTrafficPolicy.

Currently, the downstream-connection idle timeout is fixed at one hour, unless it is less than the request timeout.
I propose adding an idleTimeout parameter to the clientTrafficPolicy configuration. This parameter would allow users to specify the duration of the downstream-connection idle timeout according to their specific requirements.

Rationale:
1. **Prevention of Load Balancer Issues:** When running a proxy in an IaaS, the load balancer of that IaaS might have an idle timeout configured (e.g., GCP with an idle timeout of 600 seconds: [documentation]( https://cloud.google.com/load-balancing/docs/https/#timeouts_and_retries:~:text=Application%20Load%20Balancer%3A-,600%20seconds,-2)). 
If the idle timeout of the LB is reached, the LB closes the connection silently without notifying the proxy, causing following requests to fail. Having a configurable downstream idle timeout can prevent this issue by ensuring that Envoy's idle timeout aligns with or is shorter than the LB’s idle timeout.

2. **Saving Resources:** Configuring the downstream idle timeout helps save resources by reducing the amount needed to maintain connections.

3. **Drain Process Optimization:** In cases where Envoy needs to be drained, having a lower idle timeout enables faster termination of idle connections and speeding up the release of resources.

### PR DESCRIPTION
**Pull request by yaelSchechter**
**What type of PR is this?**
Implement idle timeout in ClientTrafficPolicy
Follow up of #3042

Fixes #2861

### CODE REVIEW

File: internal/xds/translator/listener.go [hunk 1]
```diff
@@ -241,8 +241,14 @@ func (t *Translator) addXdsHTTPFilterChain(xdsListener *listenerv3.Listener, irL
...
 	}
 
-	if irListener.Timeout != nil && irListener.Timeout.HTTP != nil && irListener.Timeout.HTTP.RequestReceivedTimeout != nil {
-		mgr.RequestTimeout = durationpb.New(irListener.Timeout.HTTP.RequestReceivedTimeout.Duration)
+	if irListener.Timeout != nil && irListener.Timeout.HTTP != nil {
+		if irListener.Timeout.HTTP.RequestReceivedTimeout != nil {
+			mgr.RequestTimeout = durationpb.New(irListener.Timeout.HTTP.RequestReceivedTimeout.Duration)
+		}
+
+		if irListener.Timeout != nil && irListener.Timeout.HTTP != nil && irListener.Timeout.HTTP.IdleTimeout != nil {
```
[the remark this answers is not in the document]
Comment (yaelSchechter) [reply internal/xds/translator/listener.go]: indeed, removed thanks

File: internal/gatewayapi/clienttrafficpolicy.go [hunk 2]
```diff
@@ -488,16 +488,19 @@ func translateClientTimeout(clientTimeout *egv1a1.ClientTimeout, httpIR *ir.HTTP
...
 				return err
 			}
-			switch {
-			case httpIR.Timeout == nil:
-				httpIR.Timeout = &ir.ClientTimeout{}
-				fallthrough
-
-			case httpIR.Timeout.HTTP == nil:
-				httpIR.Timeout.HTTP = &ir.HTTPClientTimeout{}
+			setHTTPTimeout(httpIR)
```
[the remark this answers is not in the document]
Comment (yaelSchechter) [reply internal/gatewayapi/clienttrafficpolicy.go]: changed it a bit, without this function

**Review by guydc — APPROVED** [review]

**Pull request merged**